components/clarinet-deployments/src/types.rs

```diff
@@ -215,16 +213,59 @@ pub enum TransactionSpecification {
     StxTransfer(StxTransferSpecification),
 }
 
-#[derive(Debug, PartialEq, Clone)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub struct StxTransferSpecification {
+    #[serde(with = "standard_principal_data_serde")]
     pub expected_sender: StandardPrincipalData,
+    #[serde(with = "principal_data_serde")]
     pub recipient: PrincipalData,
     pub mstx_amount: u64,
+    #[serde(with = "memo_serde")]
     pub memo: [u8; 34],
     pub cost: u64,
     pub anchor_block_only: bool,
 }
 
+pub mod principal_data_serde {
+    use clarity_repl::clarity::vm::types::PrincipalData;
+    use serde::{Deserialize, Deserializer, Serializer};
+
+    pub fn serialize<S>(x: &PrincipalData, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        s.serialize_str(&x.to_string())
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<PrincipalData, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        PrincipalData::parse(&s).map_err(serde::de::Error::custom)
+    }
+}
+
+pub mod memo_serde {
+    use serde::{Deserialize, Deserializer, Serializer};
+
+    pub fn serialize<S>(x: &[u8; 34], ser: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let s = String::from_utf8_lossy(x);
+        ser.serialize_str(&s)
+    }
+
+    pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<[u8; 34], D::Error> {
+        let s = String::deserialize(d).unwrap();
```

Comment:
I don't want to take up too much time with detailed errors here, but we can at least bubble this up to the result with this (I think, I haven't tested this so you'll want to verify). We can do this on all `.unwrap()`s in our `ser/de` functions
```
        let s = String::deserialize(d).map_err(serde::de::Error::custom)?;
```

Reply:
Done